### DISCUSSION TITLE
Largest FPGA currently supported (# of look up tables)

### DISCUSSION THREAD

**Discussion by JamesTimothyMeech** [BrunoLevy/learn-fpga#91] [General]
Thanks for this great repository! I think it will save me a lot of time in my project. 

Is the largest FPGA in terms of look up tables that this repository supports the Artix 7 35T Arty FPGA Evaluation Kit: https://www.xilinx.com/products/boards-and-kits/arty.html? 

I'm going to attempt to use this repository run some C++ programs over a custom RISC-V SoC and I want to make sure I purchase the largest FPGA that is supported out of the box.

**Comment by BrunoLevy** [comment]
Hi,
- I'd rather recommend an ULX3S (the one with a 85K LUTs ECP5), because ECP5 support in the open-source tool suite is stable and mature.  
- (for experts only) using an ARTY is still possible, but you will probably have to tinker a bit to make it work (as fas as I know, ARTY support by the open-source toolchain is not as mature as for ECP5 and Ice40. In general, it is not "out of the box" as you said). If you finally decide to take this direction, then adapting the configuration file from the 35T to the 100T will not be a big deal.
- a third possibility: you can use an ARTY with Vivado (but I really prefer to use tools that can be started by scripts rather than an heavy IDE, but it also works...)

**Reply by JamesTimothyMeech** [reply]
Thanks for the tips!

**Reply by JamesTimothyMeech** [reply]
I'll probably go with the ARTY because vendors don't want to ship the ULX3S to the UK because it is not CE marked

**Reply by BrunoLevy** [reply]
Before filing your order, I'd recommend you try to install yosys and nextpnr-xilinx, and check whether you manage to generate bitstreams for the arty (and for that you do not need to have the arty).

**Reply by JamesTimothyMeech** [reply]
Will do, thanks!

**Reply by JamesTimothyMeech** [reply]
For anyone who also tries this I found that these tools and examples can generate bitstreams for the ARTY out of the box: https://github.com/openXC7 

I was unable to set up the prjxray dependency for nextpnr-xilinx on my Ubuntu 22.04 machine.

I have not tested the bitstream on a physical ARTY board.

**Comment by tcal-x** [comment]
One small comment -- Vivado *can* be used via scripts (no GUI or IDE).   The LiteX build scripts do this, as well as the "fuzzers" that the Symbiflow group has built.    Of course we agree with the many other reasons to prefer open source tools!

**Reply by BrunoLevy** [reply]
Interesting ! will be good to compare LUT usage and fmax.
